jsettlers.logic/src/main/java/jsettlers/ai/highlevel/PioneerGroup.java

```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+ * Copyright (c) 2015
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *******************************************************************************/
+package jsettlers.ai.highlevel;
+
+import jsettlers.common.movable.EMovableAction;
+import jsettlers.common.movable.EMovableType;
+import jsettlers.common.position.ShortPoint2D;
+import jsettlers.input.tasks.ConvertGuiTask;
+import jsettlers.logic.map.grid.movable.MovableGrid;
+import jsettlers.logic.movable.Movable;
+import jsettlers.network.client.interfaces.ITaskScheduler;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * @author codingberlin
+ */
+public class PioneerGroup {
+
+	private int targetSize;
+	private List<Integer> pioneerIds;
+
+	private PioneerGroup(List<Integer> pioneerIds, int targetSize) {
+		this.targetSize = targetSize;
+		this.pioneerIds = pioneerIds;
+	}
+
+	public PioneerGroup(int targetSize) {
+		this(new ArrayList<Integer>(targetSize), targetSize);
+	}
+
+	public PioneerGroup(List<Integer> pioneerIds) {
```

Comment:
Are you sure you don't want to copy that list?
If yes, please add a note that that list may not be modified after constructing the object.